csi: move csi management to admin (#17176)
rook/rook: Documentation/Helm-Charts/csi-drivers-chart.md

---
title: Ceph-CSI Driver Helm Chart
---

To configure the Ceph-CSI drivers, Rook requires the installation of the [Ceph-CSI Driver chart](https://github.com/ceph/ceph-csi-operator/blob/main/docs/helm-charts/drivers-chart.md). This chart configures the CSI drivers to provision and mount volumes to make available the Ceph storage to your applications.

## Prerequisites

* The `rook-ceph` chart must be installed before the Ceph-CSI drivers chart, to install the required Ceph-CSI operator and CRDs.

## Installing

The Ceph-CSI drivers Helm chart installs the resources needed for [ceph-csi](https://github.com/ceph/ceph-csi) to run under the ceph-csi-operator.

The `helm install` command deploys the drivers in the default configuration from the chart. For more configuration options, see the [Ceph-CSI Drivers Configuration](https://github.com/ceph/ceph-csi-operator/blob/main/docs/helm-charts/drivers-chart.md#configuration).

> Ceph-CSI publishes the drivers chart from the `ceph-csi-operator` Helm repository.:

Reviewer: nit: also remove extra blank line before this 
```suggestion
Ceph-CSI publishes the drivers chart from the `ceph-csi-operator` Helm repository.
```

Author: Done

---

> title: Ceph-CSI Helm Charts

Reviewer: ```suggestion
title: Ceph-CSI Driver Helm Chart
```

Author: Done

---

To configure the Ceph-CSI drivers, Rook requires the installation of the [Ceph-CSI Driver chart](https://github.com/ceph/ceph-csi-operator/blob/main/docs/helm-charts/drivers-chart.md). This chart configures the CSI drivers to provision and mount volumes to make available the Ceph storage to your applications.

## Prerequisites

* The `rook-ceph` chart must be installed before the Ceph-CSI drivers chart, to install the required Ceph-CSI operator and CRDs.

## Installing

The Ceph-CSI drivers Helm chart installs the resources needed for [ceph-csi](https://github.com/ceph/ceph-csi) to run under the ceph-csi-operator.

> 1. Add the Ceph-CSI Operator chart repository and install the drivers chart.

Reviewer: This seems redundant, they can just see the steps below, I would suggest to remove this line.
```suggestion
```

> The `helm install` command deploys the drivers in the default configuration from the chart. The [configuration](#configuration) section points to the full parameter documentation upstream.

Reviewer: How about if we just link to the config here, and no need for the other section below.
```suggestion
The `helm install` command deploys the drivers in the default configuration from the chart. For more configuration options, see the [Ceph-CSI Drivers Configuration](https://github.com/ceph/ceph-csi-operator/blob/main/docs/helm-charts/drivers-chart.md#configuration).
```

> ### Release

Reviewer: I think no need for this section header. We don't have a separate release and development section in this doc.
```suggestion
```

> Ceph publishes the drivers chart from the `ceph-csi-operator` Helm repository (the same base URL as the `rook-ceph` chart dependency on `ceph-csi-operator` in [Chart.yaml](https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph/Chart.yaml)):

Reviewer: Let's simplify, no need to point out the dependency.
```suggestion
Ceph publishes the drivers chart from the `ceph-csi-operator` Helm repository.
```

```console
helm repo add ceph-csi-operator https://ceph.github.io/ceph-csi-operator

> helm install ceph-csi-drivers --create-namespace --namespace ceph-csi-driver ceph-csi-operator/ceph-csi-drivers

Reviewer: They shouldn't need to create the namespace, and we want the default in the same namespace as the operator. 
```suggestion
helm install ceph-csi-drivers --namespace rook-ceph ceph-csi-operator/ceph-csi-drivers
```

```

For example settings, see the chart’s [values.yaml](https://github.com/ceph/ceph-csi-operator/tree/main/deploy/charts/ceph-csi-drivers/values.yaml) in the Ceph-CSI Operator repository.

> ## Configuration

Reviewer: No need for this section with the suggestion above
